### PR TITLE
Drop deprecated pkg_resources

### PR DESCRIPTION
**Pull request by ajjackson**
This line was causing some deprecation warnings. I think this is the right way to modernise it?

### CODE REVIEW

**Comment by ajjackson** [comment]
Oldlibs is failing, presumably there is a (Matplotlib?) version that doesn't like the Path objects and expects a string.

Could convert to string but then it will stay that way forever: tempted to narrow down the problematic version and apply tweak in that case only. Then it is more likely to be cleaned up later...

**Comment by ajjackson** [comment]
After a bit of binary search, Matplotlib styles support Path objects from version 3.2 (March 2020). It doesn't seem unreasonable to bump the minimum requirement to this.

(Mantid is currently pinned to version 3.6)

**Comment by ajjackson** [comment]
@oerc0122 @mducle Is there any reason to be concerned about bumping the Matplotlib dependency a bit? It's nice to do things with Path objects where possible :smile:

File: CHANGELOG.rst [hunk 1]
```diff
@@ -5,13 +5,15 @@
...
   - Python 3.7 is no longer supported
 
-    - Minimum version of scipy increased from 1.1 to 1.10
+  - Minimum version of scipy increased from 1.1 to 1.10
 
-      - This requires numpy >= 1.19.5
+    - This requires numpy >= 1.19.5
 
-    - Minimum version of pint increased from 0.10.1 to 0.19
+  - Minium version of matplotlib increased from 2.2.2 to 3.2.0
```
Comment (oerc0122) [inline CHANGELOG.rst:12]: Minimum

**Review by oerc0122 — CHANGES_REQUESTED** [review]
Just a typo here.

Don't think there should be any concern about bumping.

**Comment by mducle** [comment]
I don't think so, I think we just use fairly standard Matplotlib methods which haven't changed between version 2 and 3.

**Comment by ajjackson** [comment]
Great, thanks :-) Will merge when tests pass again.

**Comment by oerc0122** [comment]
Having taken a look at the docs, the main difference is that you no longer have Python 2 support (which I don't think is an issue at this point) and unless you're mucking around in the backend, the interfaces are much the same for standard calls.

**Review by oerc0122 — APPROVED** [review]

**Pull request merged**